src/components/content/order/locks/Locks.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import React, { useRef } from 'react';
+import { Button, Form, Space, Switch } from 'antd';
+import '../../../../styles/locks.css';
+import {
+    ApiError,
+    DeployedServiceDetails,
+    Response,
+    ServiceLockConfig,
+    ServiceService,
+    VendorHostedDeployedServiceDetails,
+} from '../../../../xpanse-api/generated';
+import { useMutation } from '@tanstack/react-query';
+import LocksResult from './LocksResult';
+
+export const Locks = ({
+    currentSelectedService,
+}: {
+    currentSelectedService: DeployedServiceDetails | VendorHostedDeployedServiceDetails;
+}): React.JSX.Element => {
+    const [form] = Form.useForm();
+    const requestResult = useRef<string[]>([]);
```

Comment:
we can avoid storing this state as well. Instead `useMutationState` can be used.